### PR TITLE
Fix deepspeed docs

### PR DESCRIPTION
**Pull request by ngoquanghuy99**
# What does this PR do?

Fixes typos and misspellings in deepspeed docs.

## Who can review?

@sgugger @stas00 @LysandreJik

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for fixing!

File: docs/source/main_classes/deepspeed.mdx [hunk 1]
```diff
@@ -31,7 +31,7 @@ won't be possible on a single GPU.
 
 🤗 Transformers integrates [DeepSpeed](https://github.com/microsoft/DeepSpeed) via 2 options:
 
-1. Integration of the core DeepSpeed features via [`Trainer`]. This is everything done for you type
+1. Integration of the core DeepSpeed features via [`Trainer`]. This is everything done for your type
```
Comment (stas00) [inline docs/source/main_classes/deepspeed.mdx:34]: it was correct in the first place. Please revert.

File: docs/source/main_classes/deepspeed.mdx [hunk 2]
```diff
@@ -604,13 +604,13 @@ The following is an example configuration for ZeRO stage 2:
 **Performance tuning:**
 
 - enabling `offload_optimizer` should reduce GPU RAM usage (it requires `"stage": 2`)
-- `"overlap_comm": true` trades off increased GPU RAM usage to lower all-reduce latency. `overlap_comm` uses 4.5x
+- `"overlap_comm": true` trade offs increased GPU RAM usage to lower all-reduce latency. `overlap_comm` uses 4.5x
```
Comment (stas00) [inline docs/source/main_classes/deepspeed.mdx:607]: it was correct in the first place. Please revert.

**Review by stas00 — COMMENTED** [review]
Thank you for improving the language, @ngoquanghuy99 

There are 2 places that need to be reverted as the proposed changes were incorrect there.

Not sure why the merge was rushed in, - @sgugger - what's the best way to fix this - revert the PR commit, then revert the 2 incorrect changes and then merge again? Thanks.

update: will be fixed by https://github.com/huggingface/transformers/pull/15355

**Comment by sgugger** [comment]
I'm sorry I overlooked mistakes in those two sentences. It's true those are not fixes, though the first one is not gramatically correct.

Do as you wish to fix those, I'll make sure to stay very far away form the deepspeed file in the future.

**Comment by stas00** [comment]
I tried revert, but I thought it'd magically resume this PR and it doesn't. So if I have to make a new PR, it's much simpler to just change it directly: https://github.com/huggingface/transformers/pull/15355

**Pull request merged**